tests/sensors/test_base.py

```diff
@@ -284,62 +284,59 @@ def test_ok_with_reschedule(self, make_sensor, time_machine, task_reschedules_fo
             if ti.task_id == DUMMY_OP:
                 assert ti.state == State.NONE
 
-    def test_fail_with_reschedule(self, make_sensor, time_machine):
+    def test_fail_with_reschedule(self, make_sensor, time_machine, session):
         sensor, dr = make_sensor(return_value=False, poke_interval=10, timeout=5, mode="reschedule")
 
+        def _get_tis():
+            tis = dr.get_task_instances(session=session)
+            yield next(x for x in tis if x.task_id == SENSOR_OP)
+            yield next(x for x in tis if x.task_id == DUMMY_OP)
+
         # first poke returns False and task is re-scheduled
         date1 = timezone.utcnow()
         time_machine.move_to(date1, tick=False)
-        self._run(sensor)
-        tis = dr.get_task_instances()
-        assert len(tis) == 2
```

Comment:
It might be worth keeping this around to make sure we don't get more than 2. Maybe `_get_tis()` can become `_split_tis(ti)` or something?

Reply:
i added the len assert back